### ISSUE TITLE
CompletionFinder: can remove all options from completion

### ISSUE THREAD

**Issue by cchampet** [kislyuk/argcomplete#115]
I found the attribute __always_complete_options__ which I think was added for this purpose. But It not seems to be used any where...
Did I miss something?

Currently I fixed the case in my code, with something like this:
```python
class CustomCompletionFinder(argcomplete.CompletionFinder):   
        def filter_completions(self, completions):
            completions = super(SamCompletionFinder, self).filter_completions(completions)
            if self.always_complete_options:
                return completions
            else:
                # remove options from completion
                return [c for c in completions if not c.startswith('-')]
```

**Comment by kislyuk** [comment]
Thank you for reporting. This was breakage due to a refactoring bug by @lisongmin. I will commit a fix shortly.

**Comment by kislyuk** [comment]
Released in v0.9.0. Please test.

**Comment by cchampet** [comment]
It works perfectly.
Thanks for your quick response.
